dashboard/app/models/user.rb

```diff
@@ -1252,6 +1252,26 @@ def can_edit_password?
     encrypted_password.present?
   end
 
+  def can_edit_display_name?
+    !picture_or_word_account?
+  end
+
+  def picture_or_word_account?
+    return false if sections_as_student.empty?
+    return false if encrypted_password.present?
+    sections_as_student.all? do |section|
+      section.login_type == Section::LOGIN_TYPE_PICTURE || section.login_type == Section::LOGIN_TYPE_WORD
+    end
+  end
+
+  def should_show_email?
+    !picture_or_word_account?
+  end
+
+  def can_delete_account?
+    !picture_or_word_account?
+  end
```

Comment:
This adds four methods, where it seems one should suffice. Perhaps `sponsored?` (as suggested above)?